File: solutions/ruby/scrabble-score/1/scrabble_score.rb
LETTER_VALUE = {
  **'AEIOULNRST'.chars.map{|c| [c, 1]}.to_h,
  **'DG'.chars.map{|c| [c, 2]}.to_h,
  **'BCMP'.chars.map{|c| [c, 3]}.to_h,
  **'FHVWY'.chars.map{|c| [c, 4]}.to_h,
  **'K'.chars.map{|c| [c, 5]}.to_h,
  **'JX'.chars.map{|c| [c, 8]}.to_h,
  **'QZ'.chars.map{|c| [c, 10]}.to_h,
}.freeze

class Scrabble
  def initialize(word)
    @word = word
  end

  def score
    @word.upcase.chars.reduce(0) { |acc, letter| acc + LETTER_VALUE[letter] }
  end
end

File: solutions/python/scrabble-score/1/scrabble_score.py
score_trans_map = {
    **dict.fromkeys(['A', 'E', 'I', 'O', 'U', 'L', 'N', 'R', 'S', 'T'], 1),
    **dict.fromkeys(['D', 'G'], 2),
    **dict.fromkeys(['B', 'C', 'M', 'P'], 3),
    **dict.fromkeys(['F', 'H', 'V', 'W', 'Y'], 4),
    **dict.fromkeys(['K'], 5),
    **dict.fromkeys(['J', 'X'], 8),
    **dict.fromkeys(['Q', 'Z'], 10)
}


def compute_score(letter: str) -> int:
    return score_trans_map.get(letter)


def score(word: str) -> int:
    return sum(map(compute_score, word.upper()))
